public/components/create_report/delivery/delivery.tsx

```diff
@@ -13,60 +13,202 @@
+  EuiCheckbox,
+  EuiSpacer,
+  EuiComboBox,
+  EuiText,
+  EuiLink,
+  EuiPopover,
+  EuiListGroup,
+} from '@elastic/eui';
 import { DeliveryRecipientsBox } from './delivery_recipients_box';
-import { DeliveryChannelRadio } from './delivery_channel_radio';
+import { email_recipient_options } from './delivery_constants';
 
-const ReportDelivery = (props) => {
-  const { 
-    deliveryEmailSubject, 
-    onChangeDeliveryEmailSubject,
-    deliveryEmailBody,
-    onChangeDeliveryEmailBody
-  } = props;
+const insert_placeholder_options = [
+  {
+    label: 'Report details URL',
+    href: '#',
+    iconType: 'link',
+    size: 's',
+  },
+  {
+    label: 'Report source URL',
+    href: '#',
+    iconType: 'link',
+    size: 's',
+  },
+  {
+    label: 'File download URL',
+    href: '#',
+    iconType: 'link',
+    size: 's',
+  },
+  {
+    label: 'Report creation timestamp',
+    href: '#',
+    iconType: 'clock',
+    size: 's',
+  },
+];
 
-  return (
-    <EuiPageContent panelPaddingSize={"l"}>
-      <EuiPageHeader>
-        <EuiTitle>
-          <h2>Delivery</h2>
-        </EuiTitle>
-      </EuiPageHeader>
-      <EuiHorizontalRule/>
-      <EuiPageContentBody>
-        <EuiFormRow label="Channel" helpText="Define delivery notification channel">
-          <DeliveryChannelRadio/>
-        </EuiFormRow>
-        <EuiFormRow label="Recipients">
-          <DeliveryRecipientsBox/>
+export function ReportDelivery() {
+  const [emailCheckbox, setEmailCheckbox] = useState(false);
+  const [emailRecipients, setEmailRecipients] = useState();
+  const [emailSubject, setEmailSubject] = useState('');
+  const [emailBody, setEmailBody] = useState('');
+  const [includeReportAsAttachment, setIncludeReportAsAttachment] = useState(
+    false
+  );
+  const [insertPlaceholder, setInsertPlaceholder] = useState(false);
+
+  const onChangeEmailCheckbox = (e) => {
+    setEmailCheckbox(e.target.checked);
+  };
+
+  const onChangeEmailRecipients = (e) => {
+    setEmailRecipients(e);
+  };
+
+  const onChangeEmailSubject = (e) => {
+    setEmailSubject(e.target.value);
+  };
+
+  const onChangeEmailBody = (e) => {
+    setEmailBody(e.target.value);
+  };
+
+  const onChangeIncludeReportAsAttachment = (e) => {
+    setIncludeReportAsAttachment(e.target.checked);
+  };
+
+  const onInsertPlaceholderClick = () =>
+    setInsertPlaceholder((insertPlaceholder) => !insertPlaceholder);
+  const closeInsertPlaceholder = () => setInsertPlaceholder(false);
+
+  const onCreateEmailRecipient = (searchValue, flattenedOptions = []) => {
+    const normalizedSearchValue = searchValue.trim().toLowerCase();
+
+    if (!normalizedSearchValue) {
+      return;
+    }
+
+    const newOption = {
+      label: searchValue,
+    };
+
+    // Create the option if it doesn't exist.
+    if (
+      flattenedOptions.findIndex(
+        (option) => option.label.trim().toLowerCase() === normalizedSearchValue
+      ) === -1
+    ) {
+      email_recipient_options.push(newOption);
```

Comment:
I believe this will not have any effect as you're contacting existing  new option at line number 119

Reply:
This adds the new option- removing it throws an error when trying to add a new option to the email recipients list